### ISSUE TITLE
ngf-accept is not validating file types correctly

### ISSUE THREAD

**Issue by wlingke** [danialfarid/ng-file-upload#718]
Try uploading a non-image, non-audio, and non-video file to your demo Play with Attributes: https://angular-file-upload.appspot.com/

It won't reject it. I think it's because in line 575, you have:

```
accept = (file.type != null && file.type.toLowerCase().match(regexp)) ||
        		(file.name != null && file.name.toLowerCase().match(regexp));

//then later...
return (accept == null || accept) && (file.size == null || (file.size < fileSizeMax && file.size > fileSizeMin));
```

but accept is "null" when both file.type and file.name do not match the regexp. As a result, (accept == null) yields true and `validate` returns true.

**Comment by wlingke** [comment]
This is an easy fix so I won't add a PR but it's as simple as changing `match` to `test` I think.

**Comment by danialfarid** [comment]
Fixed in version 4.1.3
